internal/pkg/warewulfd/sentDB.go

```diff
@@ -0,0 +1,144 @@
+package warewulfd
+
+import (
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"path"
+	"sync"
+
+	"github.com/hpcng/warewulf/internal/pkg/wwlog"
+)
+
+/*
+store the sent files name and its checksum
+*/
+type SentFiles struct {
+	Files     []File `json:"files:"`
+	sha256sum [32]byte
+	Sha256hex string `json:"sha256"`
+	wwinit    bool
+}
+
+type File struct {
+	FileName  string `json:"file name"`
+	sha256sum [32]byte
+	Sha256hex string `json:"sha256"`
+}
+
+/*
+Database for the checksum of sent files, this
+values can be used for measured boot in combination with
+TPM devicces
+*/
+var sentDB map[string]*SentFiles
+
+// mutex for locking the map
+var mu sync.Mutex
+
+func init() {
+	sentDB = map[string]*SentFiles{}
```

Comment:
not sure the current use case falls into the usage of sync.map
https://pkg.go.dev/sync#Map
 (1) when the entry for a given key is only ever written once but read many times, as in caches that only grow, or 
 (2) when multiple goroutines read, write, and overwrite entries for disjoint sets of keys. In these two cases, use of a Map may significantly reduce lock contention compared to a Go map paired with a separate Mutex or RWMutex.